org.OpenT2T.Onboarding.VeraHub/node/hub.js

```diff
@@ -0,0 +1,243 @@
+    {
+        console.log("[VeraHub] -------------------------")
+        console.log("[VeraHub] makeVeraRequest: " + url)
+        console.log("[VeraHub] path           : " + path);
+        console.log("[VeraHub] sessionToken   : " + sessionToken);
+        console.log("[VeraHub] keepAlive      : " + keepAlive);
+        console.log("[VeraHub] mmsInfo        : " + mmsInfo);
+
+        var deferred = q.defer();
+
+        var getOptions = {
+            protocol: 'https:',
+            host: url,
+            path: path,
+            method: 'GET'
+        };
+
+        if (sessionToken)
+        {
+            getOptions.headers = { 'MMSSession': sessionToken };
+        }
+        else if (mmsInfo)
+        {
+            getOptions.headers = {
+                'MMSAuth': mmsInfo.Identity,
+                'MMSAuthSig': mmsInfo.IdentitySignature
+            };
+        }
+
+        if (keepAlive)
+        {
+            var keepAliveAgent = new https.Agent({keepAlive: true});
+            getOptions.agent = keepAliveAgent;
+        }
+
+        var request = https.get(getOptions, function(response)  {
+            var body = '';
+            response.setEncoding('utf8');
+            response.on('data', function(data) {
+                body += data;
+            });
+
+            response.on('end', function() {
+                if (response.statusCode != 200) {
+                    deferred.reject(response);
+                } else {
+                    if (returnRawBody)
+                    {
+                        deferred.resolve(body);
+                    }
+                    else
+                    {
+                        var parsedBody = JSON.parse(body);
+                        deferred.resolve(parsedBody);
+                    }
+                }
+            });
+
+            response.on('error', function(e) {
+                deferred.reject(e);
+            });
+        });
+
+        request.on('error', (e) => {
+            deferred.reject(e);
+        });
+
+        return deferred.promise;
+    },
+};
+
+module.exports = 
+{
+    inputNeeded : [
+        {
+            type: 'input',
+            name: 'pkDevice',
+            message: 'Vera Device Serial Number (8 digit numeric): ',
+            validate: function(value) {
+                var pass = !!value;
+                if (pass) {
+                    return true;
+                } else {
+                    return 'Please enter a valid Serial Number.';
+                }
+            }
+        },
+        {
+            type: 'input',
+            name: 'username',
+            message: 'Vera User Name (create this in the Vera app): ',
+            validate: function(value) {
+                var pass = !!value;
+                if (pass) {
+                    return true;
+                } else {
+                    return 'Please enter a valid User Name.';
+                }
+            }
+        },
+        {
+            type: 'input',
```

Comment:
Change to type to 'password' so that it is not visible on console, while entering.